### PR TITLE
chore: prepare package metadata for publishing

### PR DESCRIPTION
**Pull request by legobeat**
- rename package from `ethjs-filter` to `@metamask/ethjs-filter`
- remove `lint:staged` package script
- add `.nvmrc` set to `v12`
- npm v5+ compat: rename `prepublish` script to `prepare`
- remove unused devDependencies
- `npm dedupe; npm audit fix`
- format `CHANGELOG.md`
  - add missing releases
- add `publishConfig` to package manifest

### CODE REVIEW

File: package.json
Comment (Mrtenz) [inline package.json:1]: Missing publish configuration.

File: package.json [hunk 1]
```diff
@@ -23,15 +20,17 @@
     "lint": "npm run lint:js",
     "lint:eslint": "eslint --ignore-path .gitignore --ignore-pattern **/**.min.js",
     "lint:js": "npm run lint:eslint -- . ",
-    "lint:staged": "lint-staged",
+    "prebuild": "npm run build:clean && npm run test",
+    "prepare": "npm run build",
```
Comment (naugtur) [inline package.json:24]: I'll leave it here to make sure the decision to switch prepublish->prepare is deliberate.
`prepare` runs before publish but also when you install from a local copy or from git. `prepublishOnly` does what it says ;)  - only runs when creating a tarball to publish to npm.

Comment (mcmire) [reply package.json:24]: Right. We'd been using `prepublishOnly` prior to switching to Yarn Berry, so I am curious why we are choosing to use `prepare` here.

Comment (mcmire) [reply package.json:24]: Mainly I was worried about introducing something that people who update these packages aren't likely to be familiar with, but if we've already been using this pattern then that's fine.

**Review by mcmire — APPROVED** [review]
Looks good.

**Pull request merged**